src/ipa/rpi/controller/rpi/clock_recovery.cpp

```diff
@@ -0,0 +1,87 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (C) 2024, Raspberry Pi Ltd
+ *
+ * Camera sync control algorithm
+ */
+#include "clock_recovery.h"
+
+#include <libcamera/base/log.h>
+
+using namespace RPiController;
+using namespace libcamera;
+
+LOG_DEFINE_CATEGORY(RPiClockRec)
+
+ClockRecovery::ClockRecovery()
+{
+	initialise();
+}
+
+void ClockRecovery::initialise(unsigned int numPts, unsigned int maxJitter, unsigned int minPts)
+{
+	numPts_ = numPts;
+	maxJitter_ = maxJitter;
+	minPts_ = minPts;
+	reset();
+}
```

Comment:
Why not move this to the constructor?

Also, maybe worth having ``maxJitter`` as a duration type? Although maybe not if the below calcs use int types.

Reply:
Ah yes, I didn't get this quite right. I _meant_ to call `initialise()` after reading the parameters from the tuning file, but forgot. So I'll put that in (also, things like `maxJitter` don't need to be member variables either any more).

I'll see if making `maxJitter` a duration helps me or not...